### DISCUSSION TITLE
Merge Queues

### DISCUSSION THREAD

**Discussion by Mab879** [ComplianceAsCode/content#11116] [Ideas]
I was wondering if enabling [Merge Queue](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue#about-merge-queues) for the project maybe a good idea. The idea behind this is to prevent issues where the PR is green, but when merged the tests fail on master. In the past we have issues with this and it that seem like they could be resolved with Merge Queues. The other solutions such as requiring all PRs to be rebased to the latest `master` seems like bad idea as this repo moves too fast for that.

If you have any questions or concerns please post them here.

**Comment by marcusburghardt** [comment]
I like this idea and see only benefits with this. Actually, what would be the drawbacks of enabling Merge Queue?

**Comment by Mab879** [comment]
I don't see any downsides at moment. I'm willing hear others thoughts.

**Comment by Mab879** [comment]
I just wanted to give a quick update on this. 

I have attempt to deploy merge queues twice. I have had to back it both times due to issue with Packit. The current issue is that Packit names the tests differently from it runs them on PRs and when it runs them on the temporary branch which causes GitHub to think the tests have not been ran. All required tests for PRs must also run for the merge queue. I will looking into work around in the coming weeks.

**Comment by Mab879** [comment]
After talking with the packit folks this is currently not possible. I have created https://github.com/packit/packit-service/issues/2266 as feature reqeust for this. For now I will add back some gating tests back to `master`.
